lib/full_url_for.js

```diff
@@ -24,7 +24,7 @@ function fullUrlForHelper(path = '/') {
     // Exit if input is an external link or a data url
     if (data.hostname !== sitehost || data.origin === 'null') return path;
 
-    path = encodeURL(config.url + `/${path}`.replace(/\/{2,}/g, '/'));
+    path = encodeURL((config.url + `/${path}`).replace(/\/{2,}/g, '/'));
```

Comment:
I printed the log here, although `encodeURL` make the url right but I don't think this is the right change.

```js
    const ddd = (config.url + `/${path}`).replace(/\/{2,}/g, '/');
    console.log(ddd);
    path = encodeURL(ddd);
    console.log(path);
```

Output:
```
PS C:\Users\MrTT\Documents\GitHub\hexo-util> yarn test -f 'internal url'
yarn run v1.19.1
$ mocha -f "internal url"


  full_url_for
https:/example.com/index.html
https://example.com/index.html
https:/example.com/
https://example.com/
    √ internal url - root directory
https:/example.com/blog/index.html
https://example.com/blog/index.html
https:/example.com/blog/
https://example.com/blog/
    √ internal url - subdirectory
https:/example.com/blog/index.html
https://example.com/blog/index.html
https:/example.com/blog/
https://example.com/blog/
    √ internal url - subdirectory with trailing slash
https:/example.com/index.html
https://example.com/index.html
    √ internal url - no duplicate slash
```